pkgs/os-specific/linux/kernel/cachyos-kernels.nix

```diff
@@ -0,0 +1,312 @@
+
+      I2C_NCT6775 = module;
+      LEDS_TRIGGER_BLKDEV = module;
+      STEAMDECK = module;
+    };
+  };
+
+  # If updating this, remember to raise the version constraints in:
+  # * pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix
+  # * pkgs/os-specific/linux/v4l2loopback/default.nix
+  mainVariant = {
+    version = "6.10.2";
+    variantDescription = "Linux SCHED-EXT + BORE + Cachy Sauce Kernel by CachyOS with other patches and improvements";
+    source = {
+      url = "mirror://kernel/linux/kernel/v6.x/linux-6.10.2.tar.xz";
+      hash = "sha256-c9hSDdnLpaz8XnII52s12XQLiq44IQqSJOMuxMDSm3A=";
+    };
+    patches = [
+      {
+        name = "cachyos-base-all";
+        patch = "${cachyPatches}/6.10/all/0001-cachyos-base-all.patch";
+      }
+      {
+        name = "sched-ext";
+        patch = "${cachyPatches}/6.10/sched/0001-sched-ext.patch";
+      }
+      {
+        name = "bore-cachy-ext";
+        patch = "${cachyPatches}/6.10/sched/0001-bore-cachy-ext.patch";
+      }
+    ];
+
+    config = with lib.kernel; {
+      # CachyOS config
+      CACHY = yes;
+
+      # CPU Scheduler
+      SCHED_CLASS_EXT = yes;
+      SCHED_BORE = yes;
+      MIN_BASE_SLICE_NS = freeform "1000000";
+
+      # LLVM level
+      LTO_NONE = yes;
+
+      # Tick rate
+      HZ_300 = unset;
+      HZ_1000 = yes;
+      HZ = freeform "1000";
+
+      NR_CPUS = lib.mkOverride 60 (freeform "320");
+
+      # Tick type
+      HZ_PERIODIC = unset;
+      NO_HZ_IDLE = unset;
+      CONTEXT_TRACKING_FORCE = unset;
+      NO_HZ_FULL = yes;
+      NO_HZ = yes;
+      NO_HZ_COMMON = yes;
+      CONTEXT_TRACKING = yes;
+
+      # Preempt type
+      PREEMPT_BUILD = yes;
+      PREEMPT_NONE = unset;
+      PREEMPT_VOLUNTARY = lib.mkOverride 60 unset;
+      PREEMPT = lib.mkOverride 60 yes;
+      PREEMPT_COUNT = yes;
+      PREEMPTION = yes;
+      PREEMPT_DYNAMIC = yes;
+
+      # Enable O3
+      CC_OPTIMIZE_FOR_PERFORMANCE = unset;
+      CC_OPTIMIZE_FOR_PERFORMANCE_O3 = yes;
+
+      # Enable bbr3
+      TCP_CONG_CUBIC = lib.mkOverride 60 module;
+      DEFAULT_CUBIC = unset;
+      TCP_CONG_BBR = yes;
+      DEFAULT_BBR = yes;
+      DEFAULT_TCP_CONG = freeform "bbr";
+
+      # Select THP
+      TRANSPARENT_HUGEPAGE_MADVISE = lib.mkOverride 60 unset;
+      TRANSPARENT_HUGEPAGE_ALWAYS = lib.mkOverride 60 yes;
+
+      USER_NS = yes;
+
+      # Arch Linux RCU config
+      TREE_RCU = yes;
+      PREEMPT_RCU = yes;
+      RCU_EXPERT = yes;
+      TREE_SRCU = yes;
+      TASKS_RCU_GENERIC = yes;
+      FORCE_TASKS_RCU = unset;
+      NEED_TASKS_RCU = yes;
+      TASKS_RCU = yes;
+      FORCE_TASKS_RUDE_RCU = unset;
+      TASKS_RUDE_RCU = yes;
+      FORCE_TASKS_TRACE_RCU = unset;
+      TASKS_TRACE_RCU = yes;
+      RCU_STALL_COMMON = yes;
+      RCU_NEED_SEGCBLIST = yes;
+      RCU_FANOUT = freeform "64";
+      RCU_FANOUT_LEAF = freeform "16";
+      RCU_BOOST = yes;
+      RCU_BOOST_DELAY = freeform "500";
+      RCU_EXP_KTHREAD = unset;
```

Comment:
It's on if  `!PREEMPT_RT && NR_CPUS <= 32`. This is not a RT kernel and NR_CPUS is larger than 32, so it will be off anyways.